sys/linux/socket.txt

```diff
@@ -161,6 +177,20 @@ cmsghdr {
 	data		array[int8]
 } [align_ptr]
 
+type cmsghdr_unaligned[SOL, TYP, DATA] {
+	cmsg_len	len[parent, intptr]
+	cmsg_level	const[SOL, int32]
+	cmsg_type	const[TYP, int32]
+	data		DATA
+} [packed]
+
+# For __sock_cmsg_send
+cmsghdr_sock [
+	mark		cmsghdr_unaligned[SOL_SOCKET, SO_MARK, int32]
+	timestamping	cmsghdr_unaligned[SOL_SOCKET, SO_TIMESTAMPING_OLD, int32]
+	txtime		cmsghdr_unaligned[SOL_SOCKET, SCM_TXTIME, int64]
+]
```

Comment:
Here is another issue here, which I think compensated the previous one by luck. The union needs to be marked as `[varlen]` otherwise it will be statically sized like C unions. But cmsg's are of different size.
Since we only have fixed-size elements and the largest one is int64 which is equal to intptr on 64-bits, it magically provided necessary alignment and compensated for absence of [align_ptr].
But if we add different data types here, it will all break.